synapse/handlers/federation.py

```diff
@@ -118,12 +118,18 @@ def on_receive_pdu(self, origin, pdu, state=None, auth_chain=None):
 
         # FIXME (erikj): Awful hack to make the case where we are not currently
         # in the room work
-        is_in_room = yield self.auth.check_host_in_room(
-            event.room_id,
-            self.server_name
-        )
-        if not is_in_room and not event.internal_metadata.is_outlier():
-            logger.debug("Got event for room we're not in.")
+        if state and auth_chain and not event.internal_metadata.is_outlier():
+            is_in_room = yield self.auth.check_host_in_room(
```

Comment:
Please comment on why we need to have a state and an auth_chain here.
I assume it's because we need to have a state and auth_chain to persist?